transifex_utils/core/validators/strings_validator.py

```diff
@@ -6,42 +6,101 @@
 
 def validate(source_file, template):
     log.log_subline_bold(f"Reading source file : '{source_file}'.")
-    source_file_placeholders_dict = get_source_file_placeholders_dict(
-        source_file)
+    source_file_string_placeholders = get_source_file_string_placeholders(
+        source_file
+    )
+    source_file_plurals_placeholders = get_source_file_plurals_placeholders(
+        source_file
+    )
     for language in constants.languages.values():
         translation_file = template.replace('<lang>', language)
         if not os.path.exists(translation_file):
             log.log_warning(f"'{translation_file}' doesn't exist.")
             continue
         log.log_subline(f"Validating '{translation_file}'.")
-        validate_translation(source_file_placeholders_dict, translation_file)
+        validate_string_translation(
+            source_file_string_placeholders, translation_file
+        )
+        validate_plurals_translation(
+             source_file_plurals_placeholders, translation_file
+        )
 
 
-def validate_translation(source_file_placeholders_dict, translation_file):
+def validate_string_translation(source_file_placeholders, translation_file):
     root = ET.parse(translation_file).getroot()
-    for str_element in root.findall('string'):
-        str_name = str_element.get('name')
-        str_value = str_element.text
-        if str_name in source_file_placeholders_dict.keys():
-            for placeholder in source_file_placeholders_dict[str_name]:
-                if not placeholder in str_value:
+    for string in root.findall('string'):
+        name = string.get('name')
+        value = ''.join(string.itertext())
+        if name in source_file_placeholders.keys():
+            for placeholder in source_file_placeholders[name]:
+                if not placeholder in value:
                     log.log_error(placeholder + " doesn't exist in '" +
-                                  str_name + "'\n File: " + translation_file)
+                                  name + "'\n File: " + translation_file)
 
 
-def get_source_file_placeholders_dict(file):
+def validate_plurals_translation(source_file_placeholders, translation_file):
+    for plural in ET.parse(translation_file).getroot().findall('plurals'):
+        if plural.get('name') in source_file_placeholders:
+            validate_item(plural, source_file_placeholders, translation_file)
+
+
+def validate_item(plural, source_file_placeholders, translation_file):
+    items = plural.findall('item')
+
+    for item in items:
+        if (get_placeholders(item.text) != get_placeholders(items[0].text)):
+            log.log_error("Plural '" + plural.get('name') + "' contain items with " +
+            "different placeholder or format specifier \n File: " + translation_file)
+            return
```

Comment:
- better to match with the source file not the first item of the same translation.
- better all the errors and mismatch strings at once(work like the compiler).